x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -923,70 +916,80 @@ private void maybeStartApiKeyRemover() {
     }
 
     /**
-     * Get API keys for given realm and user name.
-     * @param realmName realm name
-     * @param userName user name
+     * Get API key(s) owned by the authenticateed user.<br>
+     * If no API key id or name is specified in the request then returns all API keys for current
+     * logged-in user.
+     *
+     * @param getMyApiKeyRequest {@link GetMyApiKeyRequest}
      * @param listener listener for {@link GetApiKeyResponse}
      */
-    public void getApiKeysForRealmAndUser(String realmName, String userName, ActionListener<GetApiKeyResponse> listener) {
+    public void getApiKeysForCurrentUser(final GetMyApiKeyRequest getMyApiKeyRequest, ActionListener<GetApiKeyResponse> listener) {
         ensureEnabled();
-        if (Strings.hasText(realmName) == false && Strings.hasText(userName) == false) {
-            logger.trace("No realm name or username provided");
-            listener.onFailure(new IllegalArgumentException("realm name or username must be provided"));
+
+        final Authentication authentication = Authentication.getAuthentication(threadPool.getThreadContext());
+        final String userName;
+        final String realmName;
+        final String apiKeyId;
+        final String apiKeyName;
+        if (authentication.getAuthenticatedBy().getType().equals("_es_api_key")) {
+            // in case of authenticated by API key, fetch key by API key id from authentication metadata
+            realmName = null;
+            userName = null;
+            apiKeyId = (String) authentication.getMetadata().get(API_KEY_ID_KEY);
+            apiKeyName = null;
```

Comment:
This looks like code that should live in the privileges layer, not in a service.